src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/Symbols/EEMethodSymbol.cs

```diff
@@ -710,10 +710,13 @@ internal override void AddSynthesizedReturnTypeAttributes(ref ArrayBuilder<Synth
 
             if (this.ReturnType.ContainsNullableReferenceTypes())
             {
-                if (SourceAssemblySymbol.GetUseSiteDiagnosticForNullableAttribute(compilation)?.Severity != DiagnosticSeverity.Error)
+                var diagnostics = DiagnosticBag.GetInstance();
+                SourceAssemblySymbol.GetUseSiteDiagnosticsForNullableAttribute(compilation, diagnostics);
```

Comment:
>SourceAssemblySymbol.GetUseSiteDiagnosticsForNullableAttribute(compilation, diagnostics); [](start = 16, length = 89)

This code looks fragile because it depends on the order of invocations for the GetUseSiteDiagnosticsForNullableAttribute method.